libbeat/tests/integration/proc_win.go

```diff
@@ -0,0 +1,120 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build windows
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+func stopCmd(p *os.Process) error {
+	return windows.GenerateConsoleCtrlEvent(windows.CTRL_BREAK_EVENT, uint32(p.Pid))
+}
+
+func getSysProcAttr() *syscall.SysProcAttr {
+	return &syscall.SysProcAttr{
+		// This disables the child from receiveing CTRL_C events
+		// But isolates other siganls from us, aka the parent
+		CreationFlags: windows.CREATE_NEW_PROCESS_GROUP,
+	}
+}
+
+// Job is wrapper for windows JobObject
+// https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects
+// This helper guarantees a clean process tree kill on job handler close
+type Job windows.Handle
+
+// CreateJobObject creates JobObject on Windows, global per process
+// Should only be initialized once in main function
+func CreateJobObject() (pj Job, err error) {
+	if pj, err = NewJob(); err != nil {
+		return pj, err
+	}
+
+	return pj, nil
+}
+
+// NewJob creates a instance of the JobObject
+func NewJob() (Job, error) {
+	h, err := windows.CreateJobObject(nil, nil)
+	if err != nil {
+		return 0, err
+	}
+
+	// From https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects
```

Comment:
This and the related code in the package was just copied from agent. It's pretty static so I don't hate it but can't say I like copying it. This could probably all just be moved to elastic-agent-libs so we don't have to clone it. We are fighting the dependency graph on this one.

Reply:
Also, you've changed the license by copying it here from ELv2 to Apache.

Reply:
:man_facepalming: I totally missed the license change...